peers/proposervm_client_test.go

```diff
@@ -0,0 +1,220 @@
+			queryParams: map[string]string{
+				"token": "token-with-dashes_and_underscores",
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			receivedParams := make(map[string]string)
+
+			server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				for key := range tt.queryParams {
+					receivedParams[key] = r.URL.Query().Get(key)
+				}
+
+				w.Header().Set("Content-Type", "application/json")
+				w.WriteHeader(http.StatusOK)
+				w.Write([]byte(`{"jsonrpc":"2.0","id":1,"result":{"height":100}}`))
+			}))
+			defer server.Close()
+
+			apiConfig := &config.APIConfig{
+				BaseURL:     server.URL,
+				QueryParams: tt.queryParams,
+			}
+			client := NewProposerVMAPI(server.URL, "test-chain", apiConfig)
+
+			ctx := context.Background()
+			client.GetProposedHeight(ctx)
+
+			for key, expectedValue := range tt.queryParams {
+				actualValue := receivedParams[key]
+				require.Equal(t, expectedValue, actualValue,
+					"Query param %s: expected %s, got %s", key, expectedValue, actualValue)
+			}
+		})
+	}
+}
+
+// TestProposerVMAPI_HTTPHeadersForwarding verifies that HTTP headers are forwarded correctly
+func TestProposerVMAPI_HTTPHeadersForwarding(t *testing.T) {
+	tests := []struct {
+		name        string
+		httpHeaders map[string]string
+	}{
+		{
+			name: "authorization header",
+			httpHeaders: map[string]string{
+				"Authorization": "Bearer test-token",
+			},
+		},
+		{
+			name: "multiple headers",
+			httpHeaders: map[string]string{
+				"Authorization": "Bearer test-token",
+				"X-API-Key":     "secret-key",
+				"X-Custom":      "custom-value",
+			},
+		},
+		{
+			name: "headers with special values",
+			httpHeaders: map[string]string{
+				"X-Token": "token-with-dashes-123",
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			receivedHeaders := make(map[string]string)
+
+			server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				for key := range tt.httpHeaders {
+					receivedHeaders[key] = r.Header.Get(key)
+				}
+
+				w.Header().Set("Content-Type", "application/json")
+				w.WriteHeader(http.StatusOK)
+				w.Write([]byte(`{"jsonrpc":"2.0","id":1,"result":{"height":100}}`))
+			}))
+			defer server.Close()
+
+			apiConfig := &config.APIConfig{
+				BaseURL:     server.URL,
+				HTTPHeaders: tt.httpHeaders,
+			}
+			client := NewProposerVMAPI(server.URL, "test-chain", apiConfig)
+
+			ctx := context.Background()
+			client.GetProposedHeight(ctx)
+
+			// Verify all headers were received.
+			for key, expectedValue := range tt.httpHeaders {
+				actualValue := receivedHeaders[key]
+				require.Equal(t, expectedValue, actualValue,
+					"Header %s: expected %s, got %s", key, expectedValue, actualValue)
+			}
+		})
+	}
+}
+
+// TestProposerVMAPI_CombinedQueryParamsAndHeaders verifies both work together
+func TestProposerVMAPI_CombinedQueryParamsAndHeaders(t *testing.T) {
```

Comment:
Is there a way that headers and query params interact that we need to test them together?